### ISSUE TITLE
[question] Artifact root in Google Cloud storage

### ISSUE THREAD

**Issue by ashrek1** [G-Research/fasttrackml#442]
I am trying to use Fastrackml for my existing Mlflow instance, I have my artifact root stored on GCS and am struggling to connect to my bucket. Is there any integrations or work arounds that anyone has found to use GCS for this?

**Comment by jgiannuzzi** [comment]
Hi @ashrek1, thanks a lot for your interest in FastTrackML and for trying it out!
Unfortunately we don't yet support GCS for storing artifacts, as no one has needed it — until now!

**Comment by jgiannuzzi** [comment]
According to the [GCS docs on interoperability](https://cloud.google.com/storage/docs/interoperability), you should be able to access your artifacts stored in GCS via our existing S3 backend.
A workaround would look thus look like this (until we have native GCS support):
* change all `gcs://` artifact locations to be `s3://` instead — this should be done in your MLFlow database _after you backed it up_
* start FastTrackML with the `--s3-endpoint-uri` parameter set to the GCS XML endpoint, and your GCS [HMAC keys](https://cloud.google.com/storage/docs/authentication/hmackeys) set to the right environment variables:
  ```
  export AWS_ACCESS_KEY_ID=ACCESS_ID
  export AWS_SECRET_ACCESS_KEY=SECRET
  fml server --s3-endpoint-uri https://storage.googleapis.com/
  ```
* your MLFlow runs can use GCS via S3 by setting them up with the following environment variables:
  ```
  export AWS_ACCESS_KEY_ID=ACCESS_ID
  export AWS_SECRET_ACCESS_KEY=SECRET
  export MLFLOW_S3_ENDPOINT_URL=https://storage.googleapis.com/
  ```

**Comment by goodala** [comment]
> Hi @ashrek1, we have implemented support for Google Cloud Storage in #472! Is this something you'd like to be able to use soon? If that's the case we could cut a new stable release with this feature!

Hi, This is something I'd be interested in using. I see that #472 has been merged already?

**Comment by jgiannuzzi** [comment]
Hi @goodala, sorry for the late reply! Support for GCS will be part of our next 0.4 release series, with 0.4.0-beta.1 planned for this week.
